### ISSUE TITLE
Vessel state change causes bugs for VesselParameterGroups

### ISSUE THREAD

**Issue by Teykn** [jrossignol/ContractConfigurator#715]
Heyo,

I was testing a contract that has 2 VesselParameterGroups, one that defines a vessel and another that uses that vessel. There’s one child parameter that is mutually exclusive between both VPGs. (VPG 1 is uncrewed, VPG 2 requires two crewmembers)

In testing, I noticed that when VPG 1 is satisfied the child parameters in VPG 2 would be completed despite the vessel not being able to satisfy all the parameters. 

tl;dr Completing a VPG that defines a vessel automatically completes all child parameters in a VPG that uses the vessel despite some child parameters not being satisfied.

**Comment by Teykn** [comment]
Did some testing, it seems that the only way to fix this is by adding completeInSequence = true in the child parameters. I'm not sure if this is intended behavior, especially since child parameters seem to be completed when they shouldn't be (e.g. 2 crew parameter being satisfied while vessel is uncrewed).

**Comment by siimav** [comment]
Possibly related? https://github.com/KSP-RO/ContractConfigurator/pull/5
